### PR TITLE
feat(sdk/extensions): Add kfp.onprem.use_k8s_secret method that puts k8s secrets in env vars

### PR DESCRIPTION
**Pull request by ConverJens**
Description of your changes:
Add task in kfp.onprem to load k8s secrets as env vars.

This is a utility for on-prem users to be able to load k8s secrets as environment variables in a pipeline. Notably, it can be used for fetching passwords for any kind of backend storage that one wishes to use, e.g. Minio etc.

**Checklist:**
- [X ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [X] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

File: sdk/python/kfp/onprem.py [hunk 1]
```diff
@@ -19,10 +21,55 @@ def _mount_pvc(task):
         return (
             task
                 .add_volume(
-                    k8s_client.V1Volume(name=volume_name, persistent_volume_claim=local_pvc)
-                )
+                k8s_client.V1Volume(name=volume_name, persistent_volume_claim=local_pvc)
```
Comment (numerology) [inline sdk/python/kfp/onprem.py:24]: Linter is acting weird here...Do you mind fixing this?

File: sdk/python/kfp/onprem.py [hunk 2]
```diff
@@ -1,3 +1,5 @@
+from typing import Dict, Text
```
Comment (numerology) [inline sdk/python/kfp/onprem.py:1]: Since we no longer support Python 2 maybe we can drop `Text`?

File: sdk/python/kfp/onprem.py [hunk 3]
```diff
@@ -19,10 +21,55 @@ def _mount_pvc(task):
...
+                k8s_client.V1VolumeMount(mount_path=volume_mount_path, name=volume_name)
+            )
         )
     return _mount_pvc
+
+
+def use_k8s_secret(secret_name: Text = 'k8s-secret', k8s_secret_key_to_env: Dict = {}):
+    """An operator that configures the container to use k8s credentials.
+
+    k8s_secret_key_to_env specifies a mapping from the name of the keys in the k8s secret to the name of the environment
```
Comment (numerology) [inline sdk/python/kfp/onprem.py:36]: nit: line too long

**Review by numerology — COMMENTED** [review]
Thanks! @ConverJens . Overall LGTM modulo some nit-picky comments.

File: sdk/python/kfp/onprem.py (hunk 1 above)
Comment (ConverJens) [reply sdk/python/kfp/onprem.py]: Pushed fix. Better?

File: sdk/python/kfp/onprem.py (hunk 2 above)
Comment (ConverJens) [reply sdk/python/kfp/onprem.py]: Pushed fix.

File: sdk/python/kfp/onprem.py (hunk 3 above)
Comment (ConverJens) [reply sdk/python/kfp/onprem.py]: Pushed fix.

**Comment by ConverJens** [comment]
@numerology Fixed nits.

**Comment by Ark-kun** [comment]
Thank you for your contribution.
/lgtm
/approve

**Pull request merged**